File: app/src/main/java/org/metabrainz/mobile/util/Resource.java
package org.metabrainz.mobile.util;

public class Resource<T> {
    private final Status status;
    private final T data;

    public Resource(Status status, T data) {
        this.status = status;
        this.data = data;
    }

    public static <S> Resource<S> getFailure(Class<S> tClass) {
        return new Resource<>(Status.FAILED, null);
    }

    public Status getStatus() {
        return status;
    }

    public T getData() {
        return data;
    }

    public enum Status {
        LOADING, FAILED, SUCCESS
    }
}

File: app/src/main/java/org/metabrainz/mobile/util/Resource.kt
package org.metabrainz.mobile.util

class Resource<T>(val status: Status, val data: T?) {

    enum class Status {
        LOADING, FAILED, SUCCESS
    }

    companion object {
        fun <S> getFailure(): Resource<S> {
            return Resource(Status.FAILED, null)
        }
    }
}